src/aiida/orm/implementation/storage_backend.py

```diff
@@ -405,9 +405,18 @@ def backup(
         :raises StorageBackupError: If an error occurred during the backup procedure.
         :raises NotImplementedError: If the storage backend doesn't implement a backup procedure.
         """
+        from aiida.common.exceptions import LockedProfileError, StorageBackupError
         from aiida.manage.configuration.settings import DEFAULT_CONFIG_FILE_NAME
+        from aiida.manage.profile_access import ProfileAccessManager
         from aiida.storage.log import STORAGE_LOGGER
 
+        # check that the AiiDA profile is not locked and request access for the duration of this backup process
+        # (locked means that possibly a maintenance operation is running that could interfere with the backup)
+        try:
+            ProfileAccessManager(self._profile).request_access()
```

Comment:
I am unfamiliar with profile locking -- does `request_access` lock the profile? If yes, where is the lock released? If no, is the locking performed later?

Reply:
Thank you for the explanation. I was essentially concerned if there is apossible  TOCTOU race here.

I guess I still don't understand how are the PID files handled. i.e. where is it ensured that these files are deleted? But that's besides the point here.

Reply:
To request a lock, it has to be done within a context manager. If the lock is granted, a lock file is written and at the end of the context manager it is automatically cleaned. When a process requests access, it writes a pid file, but since it is not inside a context manager this doesn't do the automatic cleanup. However, when a lock is requested, all existing pid files are checked and are removed if they are stale. So only downside is that pid-files can pile up between lock requests.